lib/pages/ride-flow/FlowWidgets.dart

```diff
@@ -0,0 +1,231 @@
+import 'package:flutter/material.dart';
+import 'package:carriage_rider/utils/CarriageTheme.dart';
+
+TextEditingController fromCtrl = TextEditingController();
+TextEditingController toCtrl = TextEditingController();
+
+class AlwaysDisabledFocusNode extends FocusNode {
+  @override
+  bool get hasFocus => false;
+}
+
+class FlowCancel extends StatelessWidget {
+  const FlowCancel({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: <Widget>[
+        Container(
+          child: InkWell(
+            child: Text('Cancel', style: CarriageTheme.cancelStyle),
+            onTap: () {
+              fromCtrl.clear();
+              toCtrl.clear();
+              Navigator.popUntil(context, ModalRoute.withName('/'));
+            },
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class BackText extends StatelessWidget {
+  const BackText({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.start,
+      children: <Widget>[
+        Container(
+          child: InkWell(
+            child: Text('Back', style: CarriageTheme.cancelStyle),
+            onTap: () {
+              Navigator.pop(context);
+            },
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+class SelectionButton extends StatelessWidget {
+  final Widget page;
+  final String text;
+  final GestureTapCallback onPressed;
+
+  const SelectionButton({Key key, this.page, this.text, this.onPressed})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return RawMaterialButton(
+      child: ButtonTheme(
+        minWidth: MediaQuery.of(context).size.width * 0.4,
+        height: 50.0,
+        shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(5)),
+        child: RaisedButton(
+          onPressed: () {
+            Navigator.push(
+                context, new MaterialPageRoute(builder: (context) => page));
+          },
+          elevation: 3.0,
+          color: Colors.white,
+          textColor: Colors.black,
+          child: Text(text,
+              style: TextStyle(fontSize: 17, fontWeight: FontWeight.bold)),
+        ),
+      ),
+      onPressed: onPressed,
+    );
+  }
+}
+
+class FlowBack extends StatelessWidget {
+  const FlowBack({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      child: Align(
+          alignment: Alignment.bottomCenter,
+          child: Padding(
+              padding: const EdgeInsets.only(bottom: 30.0),
+              child: Row(
+                children: <Widget>[
+                  ButtonTheme(
+                    minWidth: MediaQuery.of(context).size.width * 0.1,
```

Comment:
Button still looks a bit wide and the arrow isn't centered on my device, so I'm not sure if minWidth is the best thing to use here